renderer/src/features/registry-servers/components/dynamic-array-field.tsx

```diff
@@ -56,6 +56,16 @@ export function DynamicArrayField<TFieldValues extends FieldValues>({
     name: name as ArrayPath<TFieldValues>,
   })
 
+  const isInvalid = useCallback(
+    (idx: number) => {
+      return (
+        formState.errors[`${name}.${idx}.value`] ||
+        formState.errors[name as Path<TFieldValues>]
```

Comment:
the second line is when there is an error for the entire field instead of just 1 array item? but in this case the error will be displayed separately for each item?

:thinking:

Reply:
the form state will be reset only when a specific field is failing, but if another field is passes, the error msg will be applied only to those field that are failing. `formState.errors[name as Path<TFieldValues>]` is needed before the submit action in order to match this validation
```
formState.errors: {"allowedPorts":[{"value":{"type":"custom","message":"Port must be a number between 1 and 65535"}}]}
```

<img width="704" height="316" alt="Screenshot 2025-07-28 at 16 35 28" src="https://github.com/user-attachments/assets/d2af23f4-17c1-4f4f-9b93-e77d332bed45" />
<img width="674" height="349" alt="Screenshot 2025-07-28 at 16 35 24" src="https://github.com/user-attachments/assets/fab686b5-097d-47f2-bce2-90659d3dc058" />